src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,118 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.exceptions.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.utils.Command;
+
+/**
+ * Manages storage information for the program.
+ */
+public class Storage {
+    /**
+     * File path of save file
+     */
+    private final String filePath;
+
+    /**
+     * Creates new instance of Storage.
+     *
+     * @param filePath File path of save file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads a list of tasks from the save file.
+     *
+     * @return List of tasks.
+     * @throws IOException   If file could not be read.
+     * @throws DukeException If save file is corrupted.
+     */
+    public ArrayList<Task> load() throws IOException, DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        File dir = new File("data");
+
+        if (!dir.exists()) {
+            dir.mkdirs();
+        }
+
+        File f = new File(filePath);
+        f.createNewFile();
+        Scanner sc = new Scanner(f);
+
+        while (sc.hasNext()) {
+            String input = sc.nextLine().trim();
+
+            if (!input.equals("")) {
+                String[] tokens = input.split("\\|");
+                Command taskType = Command.valueOf(tokens[0]);
+                Task task;
+
+                String date;
+                String time;
+
+                switch (taskType) {
+                case EVENT:
+                    date = tokens[3];
+                    time = tokens.length < 5 ? "" : " " + tokens[4];
+                    task = new Event(tokens[2], date + time);
```

Comment:
Perhaps it would be good to add comments to some of your code logic such as tokens.length <5